src/App/Fossa/Config/Common.hs

```diff
@@ -302,17 +308,36 @@ collectRevisionData (BaseDir basedir) maybeConfig cacheStrategy cliOverride = do
   let override = collectRevisionOverride maybeConfig cliOverride
   case cacheStrategy of
     ReadOnly -> do
-      inferred <- inferProjectFromVCS basedir <||> inferProjectCached basedir <||> inferProjectDefault basedir
+      inferred <- inferVCSInfo $ inferProjectCached basedir <||> inferProjectDefault basedir
       pure $ mergeOverride override inferred
     WriteOnly -> do
-      inferred <- inferProjectFromVCS basedir <||> inferProjectDefault basedir
+      inferred <- inferVCSInfo $ inferProjectDefault basedir
       let revision = mergeOverride override inferred
       saveRevision revision
       pure revision
+  where
+    inferVCSInfo ::
+      ( Has Diagnostics sig m
+      , Has Logger sig m
+      , Has ReadFS sig m
+      , Has Exec sig m
+      ) =>
+      m InferredProject ->
+      m InferredProject
+    inferVCSInfo nextStep = do
+      vcsInfo <- errorBoundary $ inferProjectFromVCS basedir
+      case vcsInfo of
+        Failure emittedWarns errGroup ->
+          do
+            logDebug (renderFailure emittedWarns errGroup "")
+            logInfo "Unable to infer project revision from version control system. The project revision will be set to the current timestamp."
```

Comment:
Wanted thoughts on this logging message. Debating if I should keep 'Unable to infer project revision from version control system' because I don't want to give users the impression that our code is failing to do something as it might raise concerns and further questions from customers.

Reply:
I think this is a good question to ask and a great thing to think about. 

In this case, I would avoid calling it an "Error" since we already use that term elsewhere for problems serious enough to call into question whether we're getting the best results - our to-be-changed usage of "Error" when outputting warnings notwithstanding. This issue really doesn't affect results at all, only how we report the scan results in FOSSA. I would do something like this: "Unable to infer project revision from VCS, using current timestamp as the revision."